packages/overlayscrollbars-ngx/src/overlayscrollbars.component.ts

```diff
@@ -88,12 +89,24 @@ export class OverlayScrollbarsComponent implements OnDestroy, AfterViewInit {
   mergeEvents(originalEvents: OverlayScrollbarsComponent['events']) {
     return mergeEventListeners(
       {
-        initialized: (...args) => this.onInitialized.emit(args),
-        updated: (...args) => this.onUpdated.emit(args),
-        destroyed: (...args) => this.onDestroyed.emit(args),
-        scroll: (...args) => this.onScroll.emit(args),
+        initialized: (...args) => this.dispatchEventIfHasObservers(this.onInitialized, args),
+        updated: (...args) => this.dispatchEventIfHasObservers(this.onUpdated, args),
+        destroyed: (...args) => this.dispatchEventIfHasObservers(this.onDestroyed, args),
+        scroll: (...args) => this.dispatchEventIfHasObservers(this.onScroll, args),
       },
       originalEvents || {}
     );
   }
+
+  private dispatchEventIfHasObservers<T>(eventEmitter: EventEmitter<T>, args: T): void {
+    // `observed` is available since RxJS@7.2 because `observers` is being deprecated.
+    if ((eventEmitter as any).observed || eventEmitter.observers.length > 0) {
```

Comment:
Is this check required? - What would happen if just 

```js
this.ngZone.run(() => eventEmitter.emit(args));
```

is being executed?.. As far as I understand it shouldn't make much difference since if no observers are present nothing should happen.

Reply:
Thanks for the explanation :)